spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/catalog/FunctionAroundWrapper.java

```diff
@@ -27,33 +27,28 @@
  * If registered as bean it will be autowired into {@link FunctionInvocationWrapper}.
  * Keep in mind that it only affects imperative invocations where input is {@link Message}
  *
- * NOTE: This API is experimental and and could change without notice. It is
+ * NOTE: This API is experimental and could change without notice. It is
  * intended for internal use only (e.g., spring-cloud-sleuth)
  *
  * @author Oleg Zhurakousky
+ * @author Artem Bilan
  * @since 3.1
  */
 public abstract class FunctionAroundWrapper implements BiFunction<Object, FunctionInvocationWrapper, Object> {
 
 	@Override
 	public final Object apply(Object input, FunctionInvocationWrapper targetFunction) {
 		String functionalTracingEnabledStr = System.getProperty("spring.sleuth.function.enabled");
-		boolean functionalTracingEnabled = StringUtils.hasText(functionalTracingEnabledStr)
-				? Boolean.parseBoolean(functionalTracingEnabledStr) : true;
+		boolean functionalTracingEnabled =
+			!StringUtils.hasText(functionalTracingEnabledStr) || Boolean.parseBoolean(functionalTracingEnabledStr);
```

Comment:
This could be simplified a bit w/ the following (Boolean.parseBoolean is tolerant of null inputs):
```java
String functionalTracingEnabledStr = System.getProperty("spring.sleuth.function.enabled", "true");
boolean functionalTracingEnabled = Boolean.parseBoolean(functionalTracingEnabledStr);
```